### PR TITLE
Fix default int types for Pandas DataFrame and Series

### PR DESCRIPTION
**Pull request by mataton**
This PR fixes underlying issues with the way Pandas handles `int`s on Windows.

Please complete the following checklist:

* [x] I have read the [contribution guidelines](https://scikit.bio/contribute.html).

* [ ] I have documented all public-facing changes in the [changelog](https://github.com/scikit-bio/scikit-bio/blob/main/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/scikit-bio/scikit-bio/blob/main/LICENSE.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied.

  - **It is your responsibility to disclose** code, documentation, or other content derived from external source(s). If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] This pull request does not include code, documentation, or other content derived from external source(s).

Note: [This document](https://scikit.bio/devdoc/review.html) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

File: skbio/util/_testing.py [hunk 1]
```diff
@@ -349,6 +349,23 @@ def assert_data_frame_almost_equal(left, right, rtol=1e-5):
...
+    in comparing observed and expected data frames in Windows. This function repairs
+    the issue by converting int64 columns of a data frame into int32 in Windows.
+
+    See: https://github.com/unionai-oss/pandera/issues/726
+
+    """
+    for col in df.select_dtypes("int").columns:
+        df[col] = df[col].astype(int)
+
+    return df
```
Comment (qiyunzhu) [inline skbio/util/_testing.py:366]: This function doesn't have to return `df` -- it manipulates `df` in place. Returning `df` can actually confuse the user, who might assume that the function creates a copy while not touching the original input.

File: skbio/stats/distance/tests/test_mantel.py [hunk 2]
```diff
@@ -549,7 +552,7 @@ def test_minimal_compatible_input(self):
         # input as array_like
         obs = pwmantel((self.minx, self.miny, self.minz),
                        alternative='greater')
-        assert_data_frame_almost_equal(obs, self.exp_results_minimal)
+        assert_data_frame_almost_equal(obs, self.exp_results_minimal) #
```
Comment (qiyunzhu) [inline skbio/stats/distance/tests/test_mantel.py:555]: What does the trailing `#` do?

File: skbio/diversity/tests/test_driver.py [hunk 3]
```diff
@@ -170,19 +170,19 @@ def test_empty(self):
         # empty vector
         actual = alpha_diversity('sobs',
                                  np.array([], dtype=np.int64))
-        expected = pd.Series([0])
+        expected = pd.Series([0]).astype(int)
```
Comment (qiyunzhu) [inline skbio/diversity/tests/test_driver.py:173]: You can do `pd.Series([0], dtype=int)`, which is simpler. (Assuming it works...)

**Review by qiyunzhu — COMMENTED** [review]
@mataton Good job! I have two minor comments.

File: skbio/stats/distance/tests/test_mantel.py (hunk 2 above)
Comment (mataton) [reply skbio/stats/distance/tests/test_mantel.py]: Accidentally left it there, it's been removed with the most recent commits.

File: skbio/util/_testing.py (hunk 1 above)
Comment (mataton) [reply skbio/util/_testing.py:366]: Without the `return` statement, the tests fail with the following error:

`AssertionError: DataFrame Expected type <class 'pandas.core.frame.DataFrame'>, found <class 'NoneType'> instead`

**Review by qiyunzhu — APPROVED** [review]
Looks good! Let's merge.

**Pull request merged**